### PR TITLE
Remove potential double free of syslog

### PR DESCRIPTION
**Pull request by tajila**

### CODE REVIEW

**Comment by tajila** [comment]
jenkins test sanity xlinux jdk21,jdk17

File: runtime/vm/CRIUHelpers.cpp [hunk 1]
```diff
@@ -1880,7 +1878,6 @@ criuCheckpointJVMImpl(JNIEnv *env,
 			j9port_control(J9PORT_CTLDATA_SYSLOG_OPEN, 0);
 			setLogOptions(vm, syslogOptions);
 		}
-		j9mem_free_memory(syslogOptions);
```
Comment (JasonFengJ9) [inline runtime/vm/CRIUHelpers.cpp:1883]: There is no usage of `syslogOptions` after this, I think only `j9mem_free_memory(syslogOptions)` at L1895 needs to be removed.

Comment (tajila) [reply runtime/vm/CRIUHelpers.cpp:1883]: I want to have a singluar call to `j9mem_free_memory(syslogOptions);` so we dont get into this situation again when code is refactored

**Review by JasonFengJ9 — APPROVED** [review]
LGTM

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
JDK17 testing was aborted by someone, but the passing tests are enough.

**Pull request merged**